### PR TITLE
4498 - Add automation id feature for textarea, circlepager, emptymessage and error page

### PR DESCRIPTION
**Pull request by ericangeles**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Adds the attributes (id & automation id) for textarea, circlepager, emptymessage and manually set id for error page. Also includes e2e and documentations.

**Related github/jira issue (required)**:

Partial #4498 

**Steps necessary to review your pull request (required)**:
- Pull this branch, build, and run the app

1.) Empty Message
- Go to http://localhost:4000/components/emptymessage/example-index.html
- Check the emptymessage component root individually
- Should have the ids and automation ids
- Go to http://localhost:4000/components/emptymessage/test-button-click.html
- Check the component root
- Should have id and automation id
- Check the button element
- Should have id and automation id. Should append `btn` on both id and automation id.

2.) Circlepager
- Go to http://localhost:4000/components/circlepager/example-index.html
- Check the circlepager component root
- Should have the id and automation id
- Check the control buttons
- Should have have ids and automation ids
- Should also append `-control` and index number

3.) Textarea
- Go to http://localhost:4000/components/textarea/example-index
- Check the textarea component root individually
- Should have the ids and automation ids

4.) Error Page
- Go to http://localhost:4000/components/error-page/example-index
- Check the error-page element root
- Should have the id and automation id

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: src/components/circlepager/circlepager.js [hunk 1]
```diff
@@ -65,6 +67,8 @@ CirclePager.prototype = {
     this.activeIndex = s.startingSlide !== null &&
       s.startingSlide > -1 && s.startingSlide < this.slides.length ?
       s.startingSlide : 0;
+
+    utils.addAttributes(this.element, this, this.settings.attributes);
```
Comment (tmcconechy) [inline src/components/circlepager/circlepager.js:71]: i think we should add sequential id's to the circle buttons

File: src/components/emptymessage/emptymessage.js [hunk 2]
```diff
@@ -90,6 +92,8 @@ EmptyMessage.prototype = {
       }
     }
 
+    utils.addAttributes(this.element, this, this.settings.attributes);
```
Comment (tmcconechy) [inline src/components/emptymessage/emptymessage.js:95]: I think we might need to append id's to the empty message action buttons

**Review by tmcconechy — COMMENTED** [review]
1. Can you check off points on 4498?
1. I think the circle pager circle "buttons" should get appended id's 
1. i tin kempty message action button should get appended

**Comment by ericangeles** [comment]
I guess I covered all the things you've mentioned, @tmcconechy. Open for review again.

**Review by tmcconechy — APPROVED** [review]
Think this looks good to me now 👍

**Review by jmacaluso711 — APPROVED** [review]

**Pull request merged**